tools/clang/include/clang/SPIRV/SPIRVContext.h

```diff
@@ -27,16 +44,47 @@ class SPIRVContext {
   SPIRVContext &operator=(const SPIRVContext &) = delete;
   SPIRVContext &operator=(SPIRVContext &&) = delete;
 
-  inline uint32_t GetNextId() const;
-  inline uint32_t TakeNextId();
+  inline uint32_t getNextId() const;
+  inline uint32_t takeNextId();
+
+  /// \brief Returns the <result-id> that defines the given Type. If the type
+  /// has not been defined, it will define and store its instruction.
+  uint32_t getResultIdForType(const Type *);
+
+  /// \brief Returns the instruction that defined the given Type.
+  /// If no <result-id> has been associated with this Type (not defined yet),
+  /// the instruction will be created, registered, and returned.
+  const std::vector<uint32_t> &getInstrForType(const Type *);
```

Comment:
I still don't quite think we should have this method in SPIRVContext. Constructing concrete SPIR-V words should be the responsibility of SPIR-V module and inst builder. SPIRVContext should be only in charge of id assignment and resource allocation etc..

IMHO we just need to keep the functionality of assigning `<result-id>`s for the types in SPIRVContext, via `registerType`. `registerType` should automatically reserve a `<result-id>` for the given type if new. No need to actually constructing the SPIR-V words for the type.

How do you think?